src/Symfony/Component/Console/Style/SymfonyStyle.php

```diff
@@ -320,4 +363,47 @@ private function getTerminalWidth()
 
         return $dimensions[0] ?: self::MAX_LINE_LENGTH;
     }
+
+    private function autoPrependBlock()
+    {
+        $chars = substr($this->bufferedOutput->fetch(), -2);
+
+        if (false === $chars) {
+            return $this->newLine(); //empty history, so we should start with a new line.
+        }
+        //Prepend new line for each non PHP_EOL chars (This means no blank line was output before)
+        $this->newLine(2 - substr_count($chars, PHP_EOL));
+    }
+
+    private function autoPrependText()
+    {
+        $chars = $this->bufferedOutput->fetch();
+        //Prepend new line if last char isn't EOL:
+        if (PHP_EOL !== substr($chars, -1)) {
+            $this->newLine();
+        }
+    }
+
+    private function reduceBuffer($messages)
+    {
+        //Retrieve last inserted chars from the two latest inserted lines:
+        $fetched = $this->bufferedOutput->fetch();
+
+        if (is_string($messages)) {
+            if (false === strpos($messages, PHP_EOL)) {
+                //No EOL, should return previous + single message
+                return substr($fetched.$messages, -2);
+            }
+            //Has EOL. Split into multiples messages.
+            $messages = array_map(function ($message) {
+                return $message.PHP_EOL; //Re-add trailing EOL
```

Comment:
> This will add one more EOL than before the split, is it expected?

Actually it is. But I can't remember the exact reason why. It has something to do with questions:
![screenshot 2015-05-15 a 18 16 02](https://cloud.githubusercontent.com/assets/2211145/7656978/83193f50-fb2e-11e4-9073-293d3e75ccf4.PNG)
_Before: current version, After: removing the extra EOL_


However, you are right about an EOL terminated $messages string... Such code wouldn't perfectly render for instance:
```
 $output->write('Lorem ipsum dolor'.PHP_EOL);
 $output->note('NOTE');
 $output->write('sit amet');
```
The note block will not prepended by a blank line :/